readers/mongodb/messages.go

```diff
@@ -116,28 +134,63 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) bson.D {
 			filter = append(filter, bson.E{Key: name, Value: value})
 		case "v":
 			bsonFilter := value
-			val, ok := query["comparator"]
-			if ok {
-				switch val.(string) {
-				case readers.EqualKey:
-					bsonFilter = value
-				case readers.LowerThanKey:
-					bsonFilter = bson.M{"$lt": value}
-				case readers.LowerThanEqualKey:
-					bsonFilter = bson.M{"$lte": value}
-				case readers.GreaterThanKey:
-					bsonFilter = bson.M{"$gt": value}
-				case readers.GreaterThanEqualKey:
-					bsonFilter = bson.M{"$gte": value}
-				}
+			if val, ok := query["comparator"].(string); ok {
+				comparator := fmt.Sprintf("$%s", mongoComparators[val]) // mongoDB comparison operator
+				bsonFilter = bson.M{comparator: value}
 			}
 			filter = append(filter, bson.E{Key: "value", Value: bsonFilter})
 		case "vb":
 			filter = append(filter, bson.E{Key: "bool_value", Value: value})
 		case "vs":
-			filter = append(filter, bson.E{Key: "string_value", Value: value})
+			bsonFilter := value
+			stringValue := "string_value"
+			if val, ok := query["comparator"].(string); ok {
+				fmt.Println("val = ", val)
+				comparator := fmt.Sprintf("$%s", mongoComparators[val])
+				fmt.Println("comparator = ", comparator)
+				switch comparator {
+				case "$eq":
+					// comparator = fmt.Sprintf("$%s", mongoComparators[val]) // mongoDB comparison operator
+					bsonFilter = bson.M{comparator: value}
+				case "$gt":
+					notBsonFilter := bson.E{Key: "$ne", Value: value}
+					value = fmt.Sprintf(".*%s.*", value)
+					comparator = "$regex"
+					regexBsonFilter := bson.E{Key: comparator, Value: value}
+					/// Can use either of the two lines below
+					// bsonFilter = bson.D{notBsonFilter, regexBsonFilter}
+					bsonFilter = bson.M{notBsonFilter.Key: notBsonFilter.Value, regexBsonFilter.Key: regexBsonFilter.Value}
+				case "$gte":
+					comparator = "$regex"
+					value = fmt.Sprintf(".*%s.*", value)
+					// bsonFilter = bson.{comparator: value}
+					bsonFilter = bson.E{Key: comparator, Value: value}
+				case "$lte":
+					// comparator = "$regex"
+					// stringValue = fmt.Sprintf("/%s/", stringValue)
+					// bsonFilter = bson.M{comparator: value}
+					notBsonFilter := bson.E{Key: "$ne", Value: value}
+					value = fmt.Sprintf(".*%s.*", value)
+					comparator = "$regex"
+					regexBsonFilter := bson.E{Key: comparator, Value: value}
+					/// Can use either of the two lines below
+					// bsonFilter = bson.D{notBsonFilter, regexBsonFilter}
+					bsonFilter = bson.M{notBsonFilter.Key: notBsonFilter.Value, regexBsonFilter.Key: regexBsonFilter.Value}
```

Comment:
@AryanGodara

The below suggested query will gives the same results like `lte` operator in `postgres-reader`

```suggestion
db.messages.aggregate([
  {
    $match: {
      $expr: {
        $regexMatch: {
          input: "Active11",
          regex: "$string_value"
        }
      }
    },
    
  }
])
```